lib/dash/dash_parser.js

```diff
@@ -1431,8 +1448,18 @@ shaka.dash.DashParser = class {
       prevEnd = start + periodDuration;
     } // end of period parsing loop
 
-    // Replace previous seen periods with the current one.
-    this.lastManifestUpdatePeriodIds_ = periods.map((el) => el.id);
+    if (newPeriod) {
+      // append new period from the patch manifest
+      for (const el of periods) {
+        const periodID = el.id;
+        if (!this.lastManifestUpdatePeriodIds_.includes(periodID)) {
+          this.lastManifestUpdatePeriodIds_.push(periodID);
+        }
```

Comment:
Maybe using set instead of array could simplify this code a bit?

Reply:
I have low trust in some part of the code. i'm afraid it can be double added

Reply:
@tykus160  sorry now i get your comment.
The lastManifestUpdatePeriodIds_ is also used somewhere else by other function.
I'm not sure we should change the type

Reply:
ok, let's keep it like that